index.bs

```diff
@@ -217,48 +227,86 @@ dictionary SetHTMLOptions {
+The same configuration can be used with both safe or unsafe methods, where
+the safe methods perform an implicit {{removeUnsafe}} operation. The intent is
 that one (or a few) configurations will be built-up early on in a page's
 lifetime, and can then be used whenever needed. This allows implementations
 to pre-process configurations.
 
-The configuration object is also query-able and can return
-[=SanitizerConfig/canonical=] configuration dictionaries,
-in both safe and unsafe variants. This allows a
-page to query and predict what effect a given configuration will have, or
-to build a new configuration based on an existing one.
+The configuration object can be queried to return a configuration dictionary.
+It can also be modified directly.
 
 <pre class=idl>
 [Exposed=(Window,Worker)]
 interface Sanitizer {
-  constructor(optional SanitizerConfig config = {});
+  constructor(optional SanitizerConfig configuration = {});
+
+  // Query configuration:
   SanitizerConfig get();
-  SanitizerConfig getUnsafe();
+
+  // Modify a Sanitizer's lists and fields:
+  undefined allowElement(SanitizerElementWithAttributes element);
+  undefined removeElement(SanitizerElement element);
+  undefined replaceElementWithChildren(SanitizerElement element);
+  undefined allowAttribute(SanitizerAttribute attribute);
+  undefined removeAttribute(SanitizerAttribute attribute);
+  undefined setComments(boolean allow);
+  undefined setDataAttributes(boolean allow);
+
+  // Remove markup that executes script. May modify multiple lists:
+  undefined removeUnsafe();
 };
 </pre>
 
+A {{Sanitizer}} has an associated <dfn for="Sanitizer">configuration</dfn>, a {{SanitizerConfig}}.
+
 <div algorithm>
-The <dfn for="Sanitizer" export>constructor</dfn>(|config|)
+The <dfn for="Sanitizer" export>constructor</dfn>(|configuration|)
 method steps are:
 
-1. Store |config| in [=this=]'s [=internal slot=].
+1. Let |valid| be the return value of [=set a config|setting=] |configuration| on [=this=].
+1. If |valid| is false, then throw a {{TypeError}}.
 
 </div>
 
 <div algorithm>
-The <dfn for="Sanitizer" export>get</dfn>() method steps are:
+The <dfn for="Sanitizer" export>get</dfn>() method steps are to return the value of [=this=]'s [=Sanitizer/configuration=].
+</div>
 
-1. Return the result of [=canonicalize a configuration=] with the value of
-   [=this=]'s [=internal slot=] and true.
+<div algorithm>
+The <dfn for="Sanitizer" export>allowElement</dfn>(|element|) method steps are to [=allow an element=] with |element| and [=this=]'s [=Sanitizer/configuration=].
+</div>
+
+<div algorithm>
+The <dfn for="Sanitizer" export>removeElement</dfn>(|element|) method steps are
+to [=remove an element=] with |element| and [=this=]'s [=Sanitizer/configuration=].
+</div>
 
+<div algorithm>
+The <dfn for="Sanitizer" export>replaceElementWithChildren</dfn>(|element|) method steps are to [=replace an element=] with |element| and [=this=]'s [=Sanitizer/configuration=].
```

Comment:
Let's also call the internal operation "replace an element with children"?

Reply:
Done. Added a pronoun, too, "with its children", since this is just plain text and I think that makes it a little clearer what the operation actually does.